### ISSUE TITLE
9570 Doesn't achieve S2 Sleep

### ISSUE THREAD

**Issue by fiksion** [jaromeyer/XPS9570-Catalina#19]
Thanks for making this. It was super easy to swap from Clover.

I notice that the computer never achieves S2 sleep as Clover did. It seems like it's sleeping, but the network is still accessible and I can still remotely log into (ssh) the computer when it seems like it's sleeping: the display turns off, the fans stop. It also "wakes" too fast, which leads me to believe it's like when you select "Prevent computer from sleeping automatically..." in Energy Saver.

My problem with Clover is that it did achieve S2 sleep, but lots of things would wake up wrong so I had to put some scripts in place before and after sleep to make it somewhat manageable.

**Comment by jaromeyer** [comment]
did you apply the power management settings mentioned in the readme?

**Comment by fiksion** [comment]
Yep, including the optional line.

**Comment by jaromeyer** [comment]
hmm are you talking about hibernation/suspend to disk? thats disabled because its buggy on hacks

**Comment by fiksion** [comment]
Yes, I know S4 sleep is disabled, but you should still be able to achieve S2 sleep, a low power mode where devices/peripherals are powered off....disk, network, USB, etc. You shouldn't be able to log into the computer remotely (ssh) in the state.

**Comment by jaromeyer** [comment]
ive updated the repo to opencore 0.5.8. maybe try the latest version of the EFI again. im pretty sure that my xps is sleeping correctly

**Comment by fiksion** [comment]
Yeah, that didn't work at all. I can't boot my Mac now. Any suggestions? The boot disk no longer shows up on the boot menu.

**Comment by fiksion** [comment]
ok got it to boot...now to test sleep

**Comment by fiksion** [comment]
OK, same result. But here's more information. This only happens when it's plugged into power. When it's on battery, it will achieve S2 sleep.

**Comment by fiksion** [comment]
Try this. 

1. Open System Preferences -> Sharing 
2. Enable Remote Login
3. Get the IP address of your computer
3. Unplug power from XPS9570
4. Close the lid
5. Wait until the fans stop or wait 2 minutes
6. ping the IP address of the XPS from another computer

You should see a bunch of Request timeouts because the computer is sleeping correctly. Then open the XPS and plug in the power and try the exact same steps as above, but don't remove the power (skip step 3). You should see the network active and responding.

**Comment by fiksion** [comment]
Also, I use an App call Scenario, which allows you to run AppleScript at triggered events. Sleep and wake are two of those events. Neither of them run scripts when the 9570 is powered.

**Comment by jaromeyer** [comment]
weird... I enabled remote login and SSH is working when the laptop is running. Then I closed the lid while it was being charged by the original power adapter and there was no open ssh session. After a few  minutes I tried to ping and ssh and there was no response.
